sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.institution.controller.getOverawardsByStudent.e2e-spec.ts

```diff
@@ -0,0 +1,108 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  createFakeStudent,
+  createFakeDisbursementOveraward,
+  createFakeUser,
+  createFakeStudentAssessment,
+  createFakeApplication,
+} from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  Application,
+  AssessmentTriggerType,
+  DisbursementOveraward,
+  DisbursementOverawardOriginType,
+  Student,
+  StudentAssessment,
+  User,
+} from "@sims/sims-db";
+
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import { OverawardDetailsAPIOutDTO } from "../../models/overaward.dto";
+import { getUserFullName } from "../../../../utilities";
+
+describe("OverawardInstitutionController(e2e)-getOverawardsByStudent", () => {
+  let app: INestApplication;
+  let userRepo: Repository<User>;
+  let studentRepo: Repository<Student>;
+  let assessmentRepo: Repository<StudentAssessment>;
+  let applicationRepo: Repository<Application>;
+  let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    userRepo = dataSource.getRepository(User);
+    studentRepo = dataSource.getRepository(Student);
+    assessmentRepo = dataSource.getRepository(StudentAssessment);
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
+  });
+
+  it("Should return student overawards", async () => {
+    // Arrange
+    const user = await userRepo.save(createFakeUser());
+    const student = await studentRepo.save(createFakeStudent());
+    // Prepare the student assessment to create overaward.
+    const application = await applicationRepo.save(
```

Comment:
shouldn't we connect the application connection with the location (to make it eligible for the search?)?

Reply:
@ann-aot I was also wondering about it but for the overawards we will need to rely on the `studentId` which will make the public institutions able to see the overawards table populated with applications from other institutions.
For the E2E tests, I would not mind having an application connected to the institution and one not connected to ensure this scenario.
@JasonCTang I hope that the business request to enable the overawards for public institutions considered the above consideration.

Reply:
As per the discussion, leaving it as it is.